specification/cognitiveservices/OpenAI.Inference/models/images.tsp

```diff
@@ -0,0 +1,137 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/http";
+
+using Azure.Core;
+using TypeSpec.Http;
+
+namespace Azure.OpenAI;
+
+@doc("The desired size of the generated images. Must be one of 256x256, 512x512, or 1024x1024.")
+enum ImageSize {
+  @doc("Image size of 256x256.")
+  Size256x256: "256x256",
+
+  @doc("Image size of 512x512.")
+  Size512x512: "512x512",
+
+  @doc("Image size of 1024x1024.")
+  Size1024x1024: "1024x1024",
+}
+
+@doc("Represents the request data used to generate images.")
+model ImageGenerationOptions {
+  @doc("A description of the desired images.")
+  prompt: string;
+
+  @doc("The number of images to generate (defaults to 1).")
+  n?: int32 = 1;
+
+  @doc("The desired size of the generated images. Must be one of 256x256, 512x512, or 1024x1024 (defaults to 1024x1024).")
+  size?: ImageSize = ImageSize.Size1024x1024;
+
+  @doc("A unique identifier representing your end-user, which can help to monitor and detect abuse.")
+  user?: string;
+}
+
+@doc("The image url if successful, and an error otherwise.")
+model ImageLocation {
+  @doc("The URL that provides temporary access to download the generated image.")
+  url?: url;
+
+  @doc("The error if the operation failed.")
+  error?: Foundations.Error;
+}
+
+@doc("The result of the operation if the operation succeeded.")
+model ImageResponse {
+  @doc("A timestamp when this job or item was created (in unix epochs).")
+  created: int64;
+
+  @doc("The images generated by the operator.")
+  data: ImageLocation[];
+}
+
+@doc("The result of the operation if the operation succeeded.")
+model ImageOperationResponse {
+  @doc("The ID of the operation.")
+  id: string;
+
+  @doc("A timestamp when this job or item was created (in unix epochs).")
+  created: int64;
+
+  @doc("A timestamp when this operation and its associated images expire and will be deleted (in unix epochs).")
+  expires?: int64;
+
+  @doc("The result of the operation if the operation succeeded.")
+  result?: ImageResponse;
+
+  @doc("The status of the operation")
+  status: State;
+
+  @doc("The error if the operation failed.")
+  error?: Foundations.Error;
+}
+
+@lroStatus
+@doc("The state of a job or item.")
+enum State {
+  @doc("The operation was created and is queued to be processed in the future.")
+  notRunning,
+
+  @doc("The operation has started to be processed.")
+  running,
+
+  @lroSucceeded
+  @doc("The operation has successfully be processed and is ready for consumption.")
+  succeeded,
+
+  @lroCanceled
+  @doc("The operation has been canceled and is incomplete.")
+  canceled,
+
+  @lroFailed
+  @doc("The operation has completed processing with a failure and cannot be further consumed.")
+  failed,
+
+  @doc("The entity has been deleted but may still be referenced by other entities predating the deletion.")
+  deleted,
```

Comment:
This is problematic if it is a terminal state. Need to follow up with the service team what this state actually represents.

Reply:
deleted was just there as because we were reusing then enum definition used in authoring. It was removed in a follow-up PR